src/Core/src/Signer/SignerV4.php

```diff
@@ -56,6 +56,82 @@ public function __construct(string $scopeName, string $region)
         $this->region = $region;
     }
 
+    public function presign(Request $request, ?Credentials $credentials, ?\DateTimeInterface $expires = null): void
+    {
+        if (null === $credentials) {
+            return;
+        }
+
+        if (false === $parsedUrl = parse_url($request->getEndpoint())) {
+            throw new InvalidArgument(sprintf('The endpoint "%s" is invalid.', $request->getEndpoint()));
+        }
+        if (null !== $sessionToken = $credentials->getSessionToken()) {
+            $request->setHeader('x-amz-security-token', $sessionToken);
+        }
+
+        $this->convertPostToGet($request);
```

Comment:
Can you tell me why we do this?

Reply:
That's convert `x-www-form-urlencoded` to `query-string`. In order to presign non-rest requests (ie SQS). 
AWS does not support mixing queryString and form post. But they support Form send as GET

nb: This is a copy/paste from official AWS SDK on EC2 and RDS service. 

I'm not such a should call this for every services. It would robably break S3 Upload. I'll dig into it today.